examples/build_dict.py

```diff
@@ -32,24 +32,21 @@ def build_dict(opt):
         dictionary = DictionaryAgent(opt)
     ordered_opt = copy.deepcopy(opt)
     cnt = 0
-    for datatype in ['train:ordered', 'valid']:
-        # we use train and valid sets to build dictionary
-        ordered_opt['datatype'] = datatype
-        ordered_opt['numthreads'] = 1
-        ordered_opt['batchsize'] = 1
-        world_dict = create_task(ordered_opt, dictionary)
-        # pass examples to dictionary
-        for _ in world_dict:
-            cnt += 1
-            if cnt > opt['dict_maxexs'] and opt['dict_maxexs'] > 0:
-                print('Processed {} exs, moving on.'.format(
-                      opt['dict_maxexs']))
-                # don't wait too long...
-                break
-            world_dict.parley()
+    # we use train set to build dictionary
+    ordered_opt['datatype'] = 'train:ordered'
+    ordered_opt['numthreads'] = 1
+    ordered_opt['batchsize'] = 1
+    world_dict = create_task(ordered_opt, dictionary)
+    # pass examples to dictionary
+    for _ in world_dict:
+        cnt += 1
+        if cnt > opt['dict_maxexs'] and opt['dict_maxexs'] > 0:
+            print('Processed {} exs, moving on.'.format(opt['dict_maxexs']))
+            # don't wait too long...
+            break
+        world_dict.parley()
     print('[ dictionary built. ]')
     dictionary.save(opt['dict_file'], sort=True)
-    # print('[ num words =  %d ]' % len(dictionary))
```

Comment:
yes, but it's incredibly useful debugging tool when it IS natural, so we should have something like it..  there's another instance of it during loading btw